src/main/java/common/tileentities/GTMTE_TFFT.java

```diff
@@ -0,0 +1,758 @@
+                    .addShape(
+                            STRUCTURE_PIECE_MID,
+                            transpose(new String[][] {{"gGGGg"}, {"GfffG"}, {"GfffG"}, {"GfffG"}, {"gGGGg"}}))
+                    .addShape(
+                            STRUCTURE_PIECE_BOTTOM,
+                            transpose(new String[][] {{"ccccc"}, {"cCCCc"}, {"cCCCc"}, {"cCCCc"}, {"ccccc"}}))
+                    .addElement(
+                            'c',
+                            buildHatchAdder(GTMTE_TFFT.class)
+                                    .atLeast(Energy, Maintenance)
+                                    .casingIndex(CASING_TEXTURE_ID_1)
+                                    .dot(1)
+                                    .buildAndChain(
+                                            onElementPass(te -> te.casingAmount++, ofBlock(TFFT_FIELD, CASING_META))))
+                    .addElement(
+                            'C',
+                            buildHatchAdder(GTMTE_TFFT.class)
+                                    .casingIndex(CASING_TEXTURE_ID_1)
+                                    .atLeast(
+                                            Energy,
+                                            Maintenance,
+                                            InputHatch.or(TFFTMultiHatch.INSTANCE),
+                                            OutputHatch.or(TFFTMultiHatch.INSTANCE))
+                                    .dot(2)
+                                    .buildAndChain(
+                                            onElementPass(te -> te.casingAmount++, ofBlock(TFFT_FIELD, CASING_META))))
+                    .addElement(
+                            'G',
+                            buildHatchAdder(GTMTE_TFFT.class)
+                                    .atLeast(
+                                            InputHatch.or(TFFTMultiHatch.INSTANCE),
+                                            OutputHatch.or(TFFTMultiHatch.INSTANCE))
+                                    .casingIndex(CASING_TEXTURE_ID_2)
+                                    .dot(3)
+                                    .buildAndChain(
+                                            ofBlockUnlocalizedName("IC2", "blockAlloyGlass", 0, true),
+                                            ofBlockUnlocalizedName("Thaumcraft", "blockCosmeticOpaque", 2, false),
+                                            BorosilicateGlass.ofBoroGlassAnyTier()))
+                    .addElement(
+                            'g',
+                            ofChain(
+                                    ofBlockUnlocalizedName("IC2", "blockAlloyGlass", 0, true),
+                                    ofBlockUnlocalizedName("Thaumcraft", "blockCosmeticOpaque", 2, false),
+                                    BorosilicateGlass.ofBoroGlassAnyTier()))
+                    .addElement('f', ofChain(new IStructureElement<GTMTE_TFFT>() {
+                        @Override
```

Comment:
I think if you'd implement the structure element like here https://github.com/GTNewHorizons/GoodGenerator/blob/master/src/main/java/goodgenerator/blocks/tileEntity/NeutronActivator.java#L450 You'd getter cleaner code and also less code that is just duplicated from the base class of structurelib.

Reply:
What I mean is the code within the override. Its mostly duplicated from the overwritten method with your values injected. However functions like addTieredBlock already exist where you can specify all valid tiered blocks. If you want I could also implement this change into this PR if you have trouble following what I mean.

Reply:
Ah, I see, but tfft storage blocks can be mixed, not sure if this method works

Reply:
Sure, you can input a list of possible blocks + meta if the blocks are not the same.

Reply:
Ah I see what you mean now. Ok then I can approve now.